src/st/no3/377D4.c

```diff
@@ -5,25 +5,33 @@ void ReplaceCandleWithDrop(Entity*);
 void EntityCandleDrop(Entity*);
 void EntityCandleHeartDrop(Entity*);
 
-extern ObjectInit* g_pStObjLayout[];
+extern s8 D_801D7118;
+extern s8 D_801D711C;
+extern s8 D_801D7D3A;
+extern s8 D_801D7D3B;
 extern u8* D_80180850;
-extern u16 D_80180AAC[];
-extern u16 D_80180B00[];
-extern ObjInit2 D_80180BFC[];
-extern u8* D_80180E50[];
 extern u8 D_80180E70[];
 extern u8 D_80180E78[];
-extern u16 D_80180E80[];
 extern u8 D_80180E90[];
-extern PfnEntityUpdate PfnEntityUpdates[];
-extern s16 D_801820C4[];
-extern ObjectInit* D_801D7110;
-extern ObjectInit* D_801D7114;
-extern s8 D_801D7118;
-extern s8 D_801D711C;
+extern u8* D_80180E50[];
+extern u16 D_80180AAC[];
+extern u16 D_80180B00[];
+extern u16 D_80180E80[];
+extern s16 D_801D7D2A;
+extern u16 D_801D7D2C;
+extern s16 D_801D7D2E;
+extern s16 D_801D7D30;
+extern s16 D_801D7D32;
+extern s16 D_801D7D36;
 extern s16 D_801D7D60;
 extern s16 D_801D7D62;
+extern s16 D_801820C4[];
```

Comment:
We should aim to order the field as their offsets. That includes renames variables such as `PfnEntityUpdates`. This will helps to understand better the layout of the overlay we are trying to decompile.

Reply:
I'm affraid I'm not sure to understand what you mean by that, sorry. Could you elaborate please ?

Reply:
For example `s16 D_801D7D62;` should be after `s16 D_801820C4[];`

Reply:
from what i understand we should order strictly by variable name, so i commited this:
is that correct , or should we reorder by type first and then name of var alphabetical second ?
![image](https://user-images.githubusercontent.com/10680000/197893172-c4ed4826-e594-4784-83de-cd2b248a500e.png)